### DISCUSSION TITLE
What's the max amount of grpc connections the etcd server can handle?

### DISCUSSION THREAD

**Discussion by mig824** [etcd-io/etcd#17041] [Q&A]
We have a very large cluster that has over 12k clients connecting to it and we notice performance start to significantly degrade once it gets to around 10k connections. Is etcd able to handle this kind of load? If so, is there some options we can tweak to help the server handle this better?

**Comment by shaoqin2** [comment]
For us performance is a combination of ETCD storage size, WAL write fsync time, number of slow reads from ETCD metrics etc. I think it's hard to give recommendations with connection number alone. It will be helpful if you share more metrics

```
etcd_mvcc_db_total_size_in_bytes
etcd_server_slow_apply_total
etcd_server_slow_read_indexes_total
etcd_disk_wal_fsync_duration_seconds_bucket // in buckets 
```
